StormyHares/BangazonCLI
# Order product menu cm

* **The Name of this branch is**
OrderProductMenu-cm

* **Please check if the PR fulfills these requirements**



* **What kind of change does this PR introduce?** (Bug Fix, Feature, Document Update...)
Feature - Products can be added to a customers order
Update .- refactored SQL script for dateTime formats 

* **What Ticket is associated with this PR**
#5 


* **What is the new behavior?**  (if this is a feature change)
When the user wishes to add a product to a customers order, the active order will be found or created and products can be added to it by using the CLI

* **List all core functionality** (Must include at least one Unit Test for Each)
No new core functionality - now interacting with the database and grabbing correct information from the Active Customer / Active Order

* **Other information / Steps to Test**:
1. git fetch
1. git checkout OrderProductMenu-cm
1. open the database and test database files
1. execute the new sql script to reformat the dateTimes in both files
1. save and close both database files
1. navigate to the test directory in terminal and run dotnet test - make sure they all pass
1. navigate to the src directory in terminal and run the project with dotnet run
1. Add a customer to the system with option 1 - follow prompt and fill in with whatever data you wish
1. After returning to the main menu - select option 2 to make a customer active in the system
1. Select the customer you have just created (number 9 if you have reset your database)
1. Add a product to the system for the active customer with option 2 - follow prompts and use any data you wish as long as the Quantity is an integer above 0
1. When prompted, select N to return to the main menu
1. Select option 3 to add a product to an order - the product you created should not be show (if you have reset your database, these are the options that are available : 
```
1. Kitten Hat
2. Poodle Hat
3. Blond Merkin
4. Christmas Merkin
5. Easter Merkin
6. Plumbus
7. Meseeks Box
```
14. Select the Kitten Hat with option 1
1. There should be 45 available, add 44 to your order
1. Select the Kitten Hat again with option 1
1. There should now only be 1 available - add it to your order as well
1. The Kitten Hat should no longer appear
1. Select q to return to the feature menu
1. select q to return to the main menu
1. select 2 to make a new customer active in the system
1. Select a customer that is not the one created above
1. Select 3 to add a product to that customers active order
1. The product you added above should now be available to add to another customers cart
